modules/glean/runtime/glean-plugin.client.ts

```diff
@@ -11,10 +11,13 @@ export default defineNuxtPlugin((nuxtApp) => {
     log.info('Glean: App mounted, start initing glean')
 
     const GLEAN_APP_ID = 'mozilla-social-web'
-    const devMode = useAppConfig().env === ('dev' || 'canary' || 'preview')
-    const uploadEnabled = devMode // this will eventually be a setting that the user can toggle
+    const env = useAppConfig().env
+    const devMode = env === ('dev' || 'canary' || 'preview')
+    const userSettings = useUserSettings()
+    const allowGlean = getPreferences(userSettings.value, 'allowGlean')
+    const uploadEnabled = devMode && allowGlean
 
-    Glean.initialize(GLEAN_APP_ID, uploadEnabled, {})
+    Glean.initialize(GLEAN_APP_ID, uploadEnabled, { channel: env })
```

Comment:
1. Updated `uploadEnabled` to check for `allowGlean` user setting
2. Pass through `env` as the `channel` when initializing Glean - this is just a little bit more of identifying information (see `appChannel` in the glean [initializing](https://mozilla.github.io/glean/book/reference/general/initializing.html?highlight=initi#gleaninitializeconfiguration) docs)